### ISSUE TITLE
how to test UI locally for docker

### ISSUE THREAD

**Issue by vishalbisa** [Hygieia/Hygieia#2804]
Hi Team,
i am trying to create a docker image and deploy in cloud as a POC , hence i am looking for a way to test the dashboard locally once i deploy the docker image.
we have this command to test locally --> gulp serve --local true
but for docker we will be executing the command --> docker run -t -p 8088:80 --link hygieia-api -i hygieia-ui:latest

can anyone please let me know how to use test data for UI when API/UI/DB are containerized.

Regards,
vishal

**Comment by vishalbisa** [comment]
@tabladrum can you please help here

**Comment by vishalbisa** [comment]
@Sbrenthughes can you please help here also

**Comment by Sbrenthughes** [comment]
@vishalbisa I am not familiar with loading test data to hygieia. I will try read up but will likely be sometime before I can respond back about it.

**Comment by vishalbisa** [comment]
Thanks @Sbrenthughes  apologies for adding another issue, i am trying to use jenkins collector with open source jenkins ( https://ci.jenkins.io ), i am getting the error as MONGO_PORT is not defined attached below screenshot, also attached my properties-buildler.sh file.
i am not sure from where these parameters are fetching as i dont see those in my properties file.

![image](https://user-images.githubusercontent.com/16570454/58724123-6a002300-83f9-11e9-8fdd-01dbfdc41d6a.png)

[jenkins.txt](https://github.com/Hygieia/Hygieia/files/3242305/jenkins.txt)

Thanks in advance

**Comment by ekta-thakur** [comment]
@vishalbisa Were you able to fix this issue?
I updated collectors>build>jenkins>docker>properties-builder.sh as below and MONGO_PORT error was resolved;

**MONGO_PORT=tcp://db:27017

if [ "$MONGO_PORT" != "" ]; then**

As I am also working on similar POC to deploy Hygieia on kubernetes cluster. were you able to bring UI up? As I am getting error for UI container:
**host not found in upstream "api" in /etc/nginx/conf.d/default.conf:26**

**Comment by vishalbisa** [comment]
@ekta yes I was able to resolve the issue and my POC is completed by setting up one widget SCM using GitHub.

To troubleshoot the issue please let me know so that we can have a screen share session so that I can fix it faster

**Comment by ekta-thakur** [comment]
Hi @vishalbisa I will be available today during 10:30 AM - 4:00 PM. please let me know your availability.

**Comment by vishalbisa** [comment]
Hi @ekta-thakur currently I am free now if possible let me know if we can have a look at the issue

**Comment by ekta-papajohns** [comment]
@vishalbisa sure, how do I contact you? skype id?

**Comment by ekta-thakur** [comment]
@vishalbisa sure, how do I contact you? skype id?

**Comment by vishalbisa** [comment]
Hangouts or any Webex Link so that I can join

**Comment by ekta-thakur** [comment]
Can you try connecting to sudhanshu.srivastava17@gmail.com over hangouts?

**Comment by rsethi-papajohns** [comment]
@vishalbisa Have you created separate mongo db service and then referred that in for other containers inside the pod

**Comment by rabindersethi** [comment]
@vishalbisa Are you using separate mongo service in kubernetes and created dashboarddb in it.
Can you please share some steps which you followed to perform the deployment of hygieia on kubernetes as we are facing issues while deploying on kubernetes.

**Comment by vishalbisa** [comment]
Yes I am using separate mongo service and post that I have done a ssh inside the container and created the dashboarddb.
Another method is you need to build and create a docker image from the source code which has the script which creates the dashboard db.

You can choose any of these methods

**Comment by ekta-thakur** [comment]
@vishalbisa Myself and rabindersethi are working together on this.

As you have mentioned above to use **docker run -t -p 8088:80 --link hygieia-api -i hygieia-ui:latest** to have UI running. 
Per your pervious comment, we have done the same, have separate service for mongo but we still have DB connectivity issue. We took one collector to start with, but none of API, UI and Jenkins aren't working. 

We have created YAMLs to create deployment and service in kubernetes. Could you please share the steps you followed or how you have done the deployment in Kubernetes.

**Comment by vishalbisa** [comment]
As I said it would be better if we have a screen share session , but I have deployed it manually and not worked on YAMLs to create deployment

**Comment by vishalbisa** [comment]
To add above point after deployment of mongo.. expose the service and take the IP address and port no and put it in application.properties file for API n COLLECTOR.
It should work then
